chore: add comment annotations to values.yaml for improved user configurability

rogercoll opened the pull request.

mlunadia commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -4,21 +4,24 @@ opentelemetry-operator:
     tag: "8.16.0-SNAPSHOT"
   targetAllocator:
-    enabled: false
+    enabled: false # Enable/disable the Operator's Target allocator to decouple the service discovery and metric collection functions of Prometheus. If enabled, per-node allocation strategy must be used with this deployment. Refer to: https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator
```
We should hyperlink URLs to reduce length, these will also work in some IDE like VS code.

Refer to [Target Allocator](https://github.com/open-telemetry/opentelemetry-operator/tree/main/cmd/otel-allocator)

mlunadia commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -176,16 +195,17 @@ collectors:
   daemon:
```
Because I have context, I think I can link the names cluster and daemon to be a Collector running as Deployment which collects cluster metrics and a collector running a a DaemonSet which collects node metrics, logs and routes traces.

It might not be that intutitive for users to learn that, we should add a comment that explains the blocks of configuration. 

`# Cluster Collector configuration: This Deployment-based collector gathers telemetry at the cluster level like cluster metrics.`

`# Node Collector configuration: This DaemonSet-based collector collects node-specific telemetry.`

mlunadia commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -176,16 +195,17 @@ collectors:
       runAsUser: 0
       runAsGroup: 0
-    scrape_configs_file: ""
+    scrape_configs_file: "" # Remove this configuration option to collect Prometheus metrics from applications on the same node that have the prometheus.io/scrape=true annotation, kubernetes node metrics, and cadvisor metrics. Refer to: https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-kube-stack#scrape_configs_file-details
```
reduce to a word and hyperlink

mlunadia reviewed (changes requested):
Great work @rogercoll, much easier to understand for users with all the context you have added. Added a couple of comments.

rogercoll commented:
@mlunadia Thanks for the feedback, addressed your suggestions in https://github.com/elastic/opentelemetry/pull/33/commits/825e94e33030c838b624c8d4c41b3af8a7b9d777

gizas commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -44,60 +47,68 @@ clusterRole:
-          - ${env:ELASTIC_ENDPOINT}
-          api_key: ${env:ELASTIC_API_KEY}
+          - ${env:ELASTIC_ENDPOINT}  # API key for Elasticsearch authentication.
+          api_key: ${env:ELASTIC_API_KEY} # List of Elasticsearch endpoints.
```
Should be the other way around, this comment should go above to env:ELASTIC_ENDPOINT and vs :)

gizas commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -805,23 +849,23 @@ collectors:
     image: docker.elastic.co/observability/elastic-otel-javaagent:1.0.0
   nodejs:
-    image: docker.elastic.co/observability/elastic-otel-node:edge
+    image: docker.elastic.co/observability/elastic-otel-node:0.4.1
```
Was this done on purpose?
And why elastic-otel-dotnet:edge still has edge tag?

rogercoll replied:
> Was this done on purpose?

Yes, it was initially changed in https://github.com/elastic/opentelemetry/pull/30

>  And why elastic-otel-dotnet:edge still has edge tag?

@jackshirazi @Mpdreamz Could you help with this? Which is the latest available version we should recommend for dotnet autoinstrumentation?

rogercoll replied:
A stable elastic-otel-dotnet version has not been released yet, resolving the thread as we will be using "edge" meanwhile.

Thanks for raising this

gizas commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -44,60 +47,68 @@ clusterRole:
+          api_key: ${env:ELASTIC_API_KEY} # API key for Elasticsearch authentication.
           logs_dynamic_index:
             enabled: true
           # tls:
```
consider to repeat this to other occurencies of tls

gizas commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -805,23 +849,23 @@ collectors:
 
 instrumentation:
```
Do we have an  url with general info for instrumentation? Somewhere to explain what this block will do?
Maybe this: https://opentelemetry.io/docs/concepts/instrumentation/zero-code/

rogercoll replied:
Definitely, added in https://github.com/elastic/opentelemetry/pull/33/commits/cebe6db6c4fa5139e06985f7ee97caa6a3772666

rogercoll commented:
> nit: Shall we consider adding a note in the start sth like "for full list of available options consider https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-kube-stack/values.yaml" ?

Great point, added a top comment in https://github.com/elastic/opentelemetry/pull/33/commits/cebe6db6c4fa5139e06985f7ee97caa6a3772666

gizas commented:
Roger reminder also for the limits,

In daemonset to have: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-kube-stack/values.yaml#L448-L454

And in deployment: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-kube-stack/values.yaml#L519-L525

To have both commented. WDYT?

rogercoll commented:
> Roger reminder also for the limits,
> 
> In daemonset to have: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-kube-stack/values.yaml#L448-L454
> 
> And in deployment: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-kube-stack/values.yaml#L519-L525
> 
> To have both commented. WDYT?

Good point, added in https://github.com/elastic/opentelemetry/pull/33/commits/f07c3772690e5e9a2d3b5e80b8b6753ed7743b72

Noticed that the `daemon` Pod has a lower memory limit in the default Chart config, I modified the values file to be aligned for our deployment. Our daemonset configuration it is more stressed (in terms of receivers/processors) than the cluster on. WDYT?

gizas commented:
> Our daemonset configuration it is more stressed (in terms of receivers/processors) than the cluster on. WDYT?

Looks fine for now. Without any actual load test can not tell for what should be the suggested values, but all ok for now

gizas commented:
Approving for docs changes! LGTM

@rogercoll please verify https://github.com/elastic/opentelemetry/pull/33#discussion_r1812823913 before merging

mlunadia commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -44,65 +50,82 @@ clusterRole:
 # Cluster Name is detected automatically for EKS/GKE/AKS
```
>Cluster Name is detected automatically for EKS/GKE/AKS 

Cluster Name is automatically detected for EKS/GKE/AKS, add the below value when cluster name is not available.

mlunadia commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -180,19 +213,31 @@ collectors:
             - elasticsearch/otel
+
+  #  Daemon is a K8s daemonset EDOT collector focused on gathering telemetry at
+  #  node level and expose an OTLP endpoint for data ingestion.
```
exposing

mlunadia replied:
> OTLP endpoint for data ingestion.

not sure about the below (?)
OTLP endpoint for data ingestion. SDKs using auto-instrumentation will use this

mlunadia commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -180,19 +213,31 @@ collectors:
       runAsUser: 0
       runAsGroup: 0
-    scrape_configs_file: ""
+    scrape_configs_file: "" # Refer to: https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-kube-stack#scrape_configs_file-details
```
hyperlink

rogercoll commented:
@mlunadia Thanks for the review, suggestions applied in https://github.com/elastic/opentelemetry/pull/33/commits/10581dbe5768371d0a933dd9405635675d673aa3

eedugon reviewed (approved):
Looks great that we are adding comments and explanations in the `values` file.
I have added some minor suggestions for your consideration.

One quick question: Is it common to add `markdown style` links in Yaml files? I'm asking because you are using the syntax [link_name] (http://destination) in multiple places.

eedugon commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -1,24 +1,28 @@
+# For advanced configuration options, refer to the [official OpenTelemetry Helm chart](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-kube-stack/values.yaml)
```
Are markdown style links common in Yaml files?

rogercoll replied:
Not sure, but it seems they are really handy when rendered with VS Code. @mlunadia might have more context

eedugon reviewed (commented):
Sorry, i think I've done two reviews by mistake :)
I have added some suggestions, let me know your thoughts.

Anyway the comments look great!

rogercoll commented on resources/kubernetes/operator/helm/values.yaml:
```diff
@@ -180,19 +212,32 @@ collectors:
+      requests:
+        cpu: 100m
+        memory: 500Mi
     presets:
```
Good catch, I commented this out because the deployment already has these resource values: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-kube-stack/values.yaml#L519-L525
see also https://github.com/elastic/opentelemetry/pull/33#issuecomment-2437268504
